source/scone/output/types/color.d

```diff
@@ -55,111 +55,38 @@ unittest
     assert(Color.whiteDark - Color.blackDark == 7);
 }
 
-/// both `ForegroundColor` and `BackgroundColor` work the same way. this is not not have duplicate code :)
-private template ColorTemplate()
-{
-    this(Color c)
-    {
-        color = c;
-    }
-
-    Color color;
-    alias color this;
-}
-
-/// Container for foreground colors
-struct ForegroundColor
-{
-    mixin ColorTemplate;
-}
-
-/// Container for background colors
-struct BackgroundColor
-{
-    mixin ColorTemplate;
-}
-
-/**
- * Example:
- * ---
- * window.write(0,0, Color.red.foreground, "string");
- * ---
- */
-ForegroundColor foreground(Color color)
```

Comment:
Could these be kept, maybe deprecated, which returns a `TextStyle`

```d
TextStyle foreground(Color color)
{
    return TextStyle().fg(color);
}

TextStyle background(Color color)
{
    return TextStyle().bg(color);
}
```

Reply:
Maybe this should be here, since I sometimes prefer to do `Color.red.foreground`.